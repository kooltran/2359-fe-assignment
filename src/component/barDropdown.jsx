import React from "react";
import PropTypes from "prop-types";
import Select from "react-select";
import { connect } from "react-redux";
import { selectProgressAction } from "../redux/actions/selectProgressAction";

const customStyles = {
  menu: provided => ({
    ...provided,
    margin: 0
  }),
  menuList: provided => ({
    ...provided,
    padding: 0
  })
};

export class BarDropDown extends React.Component {
  static propTypes = {
    bars: PropTypes.array.isRequired,
    selectProgressAction: PropTypes.func.isRequired
  };

  handleChangeProgress = ({ value }) => {
    this.props.selectProgressAction(value);
  };

  generateProgressBarDropdown = () => {
    const { bars } = this.props;
    return bars.map((bar, idx) => ({
      value: idx + 1,
      label: `Progress ${idx + 1}`
    }));
  };

  render() {
    const options = this.generateProgressBarDropdown();
    return (
      <Select
        className="progress-dropdown"
        options={options}
        isSearchable={false}
        defaultValue={options[0]}
        styles={customStyles}
        onChange={this.handleChangeProgress}
      />
    );
  }
}

const mapDispatchToProps = {
  selectProgressAction
};

const mapStateToProps = state => {
  return {
    bars: state.barsInfoReducer.data.bars
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(BarDropDown);
