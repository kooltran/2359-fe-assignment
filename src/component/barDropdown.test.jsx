import React from "react";
import { shallow } from "enzyme";
import { BarDropDown } from "./barDropdown";
import Select from "react-select";

describe("renders BarDropDown component", () => {
  const baseProps = {
    bars: [10, 20, 30],
    selectProgressAction: jest.fn()
  };

  it("should render properly with base props", () => {
    const wrapper = shallow(<BarDropDown {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("selectProgressAction should be call bar when change dropdown value", () => {
    const wrapper = shallow(<BarDropDown {...baseProps} />);
    wrapper.find(Select).simulate("change", { value: 10 });
    expect(baseProps.selectProgressAction).toHaveBeenCalledWith(10);
  });
});
