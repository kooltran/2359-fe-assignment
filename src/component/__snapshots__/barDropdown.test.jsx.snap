// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BarDropDown component should render properly with base props 1`] = `
<StateManager
  className="progress-dropdown"
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={
    Object {
      "label": "Progress 1",
      "value": 1,
    }
  }
  isSearchable={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Progress 1",
        "value": 1,
      },
      Object {
        "label": "Progress 2",
        "value": 2,
      },
      Object {
        "label": "Progress 3",
        "value": 3,
      },
    ]
  }
  styles={
    Object {
      "menu": [Function],
      "menuList": [Function],
    }
  }
/>
`;
